include/fuse.h

```diff
@@ -1004,6 +1004,17 @@ int fuse_getgroups(int size, gid_t list[]);
 int fuse_interrupted(void);
 
 /**
+ * Invalidates cache for the given path.
+ *
+ * This calls fuse_lowlevel_notify_inval_inode internally.
+ * This only does something for nodes which are present in
+ * the hashes; it is a no-op for paths to forgotten nodes.
+ *
+ * @return 0 on success, negative error value otherwise.
```

Comment:
I followed the original code by Miklos here, and I think the high-level file system does not need to be concerned with nodes being forgotten by FUSE, and so the invalidating routine should not return an error if the node to be is not present in the hashes. If I understand it correctly, if a node is not present in the caches (and FUSE hashes), the operating system does not care about the caches being invalidated. Please note that `lookup_path_in_cache` never does an actual lookup with the user-space file system, so returning `ENOENT` from here could actually be misleading.

An example situation to show this might be where in a network file system with a centralized server, the server notifies the client (which maps the server storage into a FUSE-based file system) about a change in an inode which the client has not "seen" yet, and hence it's not present in the caches & hashes. My feeling is that there should be no error for the client in that case if it passes the notification as a cache invalidation request to `libfuse`.